src/tree/updater_quantile_hist.cc

```diff
@@ -1182,24 +1239,26 @@ void QuantileHistMaker::Builder::InitNewNode(int nid,
 
   {
     auto& stats = snode_[nid].stats;
-    GHistRow hist = hist_[nid];
+    GHistRowT hist = hist_[nid];
+    GradientPairT grad_stat;
     if (tree[nid].IsRoot()) {
       if (data_layout_ == kDenseDataZeroBased || data_layout_ == kDenseDataOneBased) {
         const std::vector<uint32_t>& row_ptr = gmat.cut.Ptrs();
         const uint32_t ibegin = row_ptr[fid_least_bins_];
         const uint32_t iend = row_ptr[fid_least_bins_ + 1];
         auto begin = hist.data();
         for (uint32_t i = ibegin; i < iend; ++i) {
-          const GradStats et = begin[i];
-          stats.Add(et.sum_grad, et.sum_hess);
```

Comment:
The stats variable defined in line 1241 is no longer used.  Delete it.

Reply:
deleted. Thanks

Reply:
But I don't think that we should fix:
`/home/fis/Workspace/XGBoost/xgboost/rabit/include/rabit/./internal/rabit-inl.h:264:16: warning: ‘void* memcpy(void*, const void*, size_t)’ writing to an object of non-trivially copyable type ‘class xgboost::detail::GradientPairInternal<float>’; use copy-assignment or copy-initialization instead [-Wclass-memaccess]
     std::memcpy(&tdst, pdst + (i * kUnit), sizeof(tdst));`

1) `class GradientPairInternal` is trivially copyable (only two data fields are presented: `grad_`, `hess_` and it's not a pointers).
2) and to fix it we should change common `ReducerSafe_`

Reply:
possibly similar warnings were for `GradStats`, here
`rabit::Reducer<GradStats, GradStats::Reduce> histred_;`

there is also two fields only:
 ```
 /*! \brief sum gradient statistics */
  GradType sum_grad { 0 };
  /*! \brief sum hessian statistics */
  GradType sum_hess { 0 };
```